test/integration/httpInfrastructureRest.spec.ts

```diff
@@ -0,0 +1,873 @@
+      assert.equal(response.status, "204");
+    });
+
+    it("should work for get200", async () => {
+      const response = await client.httpSuccess.get200();
+      assert.equal(response.status, "200");
+    });
+
+    it("should work for head200", async () => {
+      const response = await client.httpSuccess.head200();
+      assert.equal(response.status, "200");
+    });
+
+    it("should work for head204", async () => {
+      const response = await client.httpSuccess.head204();
+      assert.equal(response.status, "204");
+    });
+
+    it("should work for head404", async () => {
+      const response = await client.httpSuccess.head404();
+      assert.equal(response.status, "404");
+    });
+
+    it("should work for patch200", async () => {
+      const response = await client.httpSuccess.patch200();
+      assert.equal(response.status, "200");
+    });
+
+    it("should work for patch202", async () => {
+      const response = await client.httpSuccess.patch202();
+      assert.equal(response.status, "202");
+    });
+
+    it("should work for patch204", async () => {
+      const response = await client.httpSuccess.patch204();
+      assert.equal(response.status, "204");
+    });
+
+    it("should work for post200", async () => {
+      const response = await client.httpSuccess.post200();
+      assert.equal(response.status, "200");
+    });
+
+    it("should work for post201", async () => {
+      const response = await client.httpSuccess.post201();
+      assert.equal(response.status, "201");
+    });
+
+    it("should work for post202", async () => {
+      const response = await client.httpSuccess.post202();
+      assert.equal(response.status, "202");
+    });
+
+    it("should work for post204", async () => {
+      const response = await client.httpSuccess.post204();
+      assert.equal(response.status, "204");
+    });
+
+    it("should work for put200", async () => {
+      const response = await client.httpSuccess.put200();
+      assert.equal(response.status, "200");
+    });
+
+    it("should work for put201", async () => {
+      const response = await client.httpSuccess.put201();
+      assert.equal(response.status, "201");
+    });
+
+    it("should work for put202", async () => {
+      const response = await client.httpSuccess.put202();
+      assert.equal(response.status, "202");
+    });
+
+    it("should work for put204", async () => {
+      const response = await client.httpSuccess.put204();
+      assert.equal(response.status, "204");
+    });
+
+    it("should work for delete204", async () => {
+      const response = await client.httpSuccess.delete204();
+      assert.equal(response.status, "204");
+    });
+  });
+
+  describe("Failure scenarios", () => {
+    it("Should get empty should throw error", async () => {
+      try {
+        const result = await client.httpFailure.getEmptyError();
+        assert.equal(result.status, "400");
+      } catch (error) {
+        console.log(error);
```

Comment:
I don't think we'll ever throw on 400 right?

Reply:
Yeah that's where I get confused, should I just check the response status ?

Reply:
Yup